spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/AggregateJobQueryDao.java

```diff
@@ -54,4 +56,7 @@ public interface AggregateJobQueryDao {
 	JobInstanceExecutions getJobInstanceExecution(String jobName, long instanceId);
 
 	JobInstanceExecutions getJobInstanceExecutions(long id, String schemaTarget);
+
+	JobInstance getJobInstance(long id, String schemaTarget) throws NoSuchJobInstanceException;
```

Comment:
I know we have passed the `schemaTarget` as string so far but I think it would be more robust and readable to simply pass in the `AppBootSchemaVersion`.  This became apparent to me when I saw the conversion to `"boot2|3"` in `SimpleJobService`.  As a user of the DAO how do I know what value to pass in here? The enum controls this.

Reply:
I'll add this as an issue to 5533, since we have to make this change because other calls use this same pattern.